packages/commons/ir-utils/src/examples/generator/generateWebSocketExample.ts

```diff
@@ -0,0 +1,225 @@
+import {
+    ExampleHeader,
+    ExamplePathParameter,
+    ExampleQueryParameterShape,
+    ExampleWebSocketMessageBody,
+    ExampleWebSocketSession,
+    HttpHeader,
+    IntermediateRepresentation,
+    PathParameter,
+    TypeDeclaration,
+    TypeId,
+    WebSocketChannel,
+    WebSocketMessage
+} from "@fern-api/ir-sdk";
+
+import { ExampleGenerationResult } from "./ExampleGenerationResult";
+import { generateTypeReferenceExample } from "./generateTypeReferenceExample";
+import { isOptional } from "./isTypeReferenceOptional";
+
+export declare namespace generateWebSocketExample {
+    interface Args {
+        ir: Omit<IntermediateRepresentation, "sdkConfig" | "subpackages" | "rootPackage">;
+        channel: WebSocketChannel;
+        typeDeclarations: Record<TypeId, TypeDeclaration>;
+        skipOptionalRequestProperties: boolean;
+    }
+
+    interface ParameterGroup<T, K> {
+        params: T[];
+        add: (example: K) => void;
+    }
```

Comment:
nit: Thoughts on naming the generic types so that they're a little more clear at face value?
```suggestion
    interface ParameterGroup<ParamT, ExampleT> {
        params: ParamT[];
        add: (example: ExampleT) => void;
    }
```